spring-cloud-skipper-autoconfigure/src/main/java/org/springframework/cloud/skipper/server/autoconfigure/ProfileApplicationListener.java

```diff
@@ -97,6 +101,19 @@ public void onApplicationEvent(ApplicationEnvironmentPreparedEvent event) {
 		}
 	}
 
+	private boolean ignoreFromSystemProperty() {
+		return Boolean.getBoolean(IGNORE_PROFILEAPPLICATIONLISTENER_PROPERTY_NAME);
+	}
+
+	private boolean ignoreFromEnvironmentVariable() {
+		boolean ignore = false;
+		String ignoreString = System.getenv(IGNORE_PROFILEAPPLICATIONLISTENER_ENVVAR_NAME);
+		if (ignoreString != null && ignoreString.length() > 0) {
```

Comment:
I prefer to use `org.springframework.util.StringUtils.isEmpty(Object)` for these checks. But I think you can just pass stuff from env to `Boolean.parseBoolean` as it handles null. Though it's a bit strange that `Boolean.getBoolean` catches NPE when it uses `Boolean.parseBoolean`.

Reply:
I'll remove the guard as indeed `Boolean.parseBoolean` handles null.  The test failure was for not cleaning up the mock it seems...it is a bit odd since I had the tests passing before submitting the PR....